Mage.Sets/src/mage/cards/s/ShareTheSpoils.java

```diff
@@ -0,0 +1,316 @@
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        Card card = game.getCard(sourceId);
+        if (card == null) { return false; }
+
+        // Have to play on your turn
+        if (!game.getActivePlayerId().equals(affectedControllerId)) { return false; }
+
+        // Not in exile
+        if (game.getState().getZone(card.getMainCard().getId()) != Zone.EXILED) { return false; }
+
+        // TODO: This is a workaround for #8706, remove when that's fixed.
+        int zoneChangeCounter = game.getState().getZoneChangeCounter(source.getSourceId());
+        // Not a card exiled with this Share the Spoils
+        ExileZone exileZone = game.getExile().getExileZone(CardUtil.getExileZoneId(game, source.getSourceId(), zoneChangeCounter));
+        if (exileZone == null) { return false; }
+        if (!exileZone.contains(sourceId)) { return false; }
+
+        ShareTheSpoilsWatcher watcher = game.getState().getWatcher(ShareTheSpoilsWatcher.class);
+
+        return watcher.hasNotUsedAbilityThisTurn(new MageObjectReference(source));
+    }
+
+    @Override
+    public ShareTheSpoilsPlayExiledCardEffect copy() { return new ShareTheSpoilsPlayExiledCardEffect(this); }
+
+    @Override
+    public boolean apply(Game game, Ability source) { return true; }
+}
+
+//-- Spend mana as any color --//
+class ShareTheSpoilsSpendAnyManaEffect extends AsThoughEffectImpl implements AsThoughManaEffect {
+
+    ShareTheSpoilsSpendAnyManaEffect() {
+        super(AsThoughEffectType.SPEND_OTHER_MANA, Duration.WhileOnBattlefield, Outcome.Benefit);
+    }
+
+    private ShareTheSpoilsSpendAnyManaEffect(final ShareTheSpoilsSpendAnyManaEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) { return true; }
+
+    @Override
+    public ShareTheSpoilsSpendAnyManaEffect copy() { return new ShareTheSpoilsSpendAnyManaEffect(this); }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        Card card = game.getCard(sourceId);
+        if (card == null) { return false; }
+
+        // TODO: This is a workaround for #8706, remove when that's fixed.
+        int zoneChangeCounter = game.getState().getZoneChangeCounter(source.getSourceId());
+        // Check Exile
+        ExileZone exileZone = game.getExile().getExileZone(CardUtil.getExileZoneId(game, source.getSourceId(), zoneChangeCounter));
+        if (exileZone == null) { return false; }
+        if (exileZone.contains(sourceId)) { return true; }
+
+        // Check Stack
+        UUID cardID;
+        Card mainCard = card.getMainCard();
+        if (mainCard instanceof ModalDoubleFacesCard) {
+            cardID = ((ModalDoubleFacesCard) mainCard).getLeftHalfCard().getId();
+        } else {
+            cardID = mainCard.getId();
+        }
+
+        CardState cardState = game.getLastKnownInformationCard(cardID, Zone.EXILED);
```

Comment:
Why it here? Will be broken on multiple movement of the same card (card's lki reset on empty stack only, so you can counter/cast same object multiple times with diff results -- first time cardState will be empty, but second time it will be non empty). If you need to check current zone then use gameState and getZone.